internal/storages/s3/folder.go

```diff
@@ -0,0 +1,185 @@
+package s3
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/pkg/errors"
+	"github.com/wal-g/wal-g/internal/storages/storage"
+	"io"
+	"strings"
+)
+
+const (
+	NotFoundAWSErrorCode  = "NotFound"
+	NoSuchKeyAWSErrorCode = "NoSuchKey"
+
+	EndpointSetting          = "AWS_ENDPOINT"
+	RegionSetting            = "AWS_REGION"
+	ForcePathStyleSetting    = "AWS_S3_FORCE_PATH_STYLE"
+	AccessKeyIdSetting       = "AWS_ACCESS_KEY_ID"
+	SecretAccessKeySetting   = "AWS_SECRET_ACCESS_KEY"
+	SseSetting               = "S3_SSE"
+	SseKmsIdSetting          = "S3_SSE_KMS_ID"
+	StorageClassSetting      = "S3_STORAGE_CLASS"
+	UploadConcurrencySetting = "UPLOAD_CONCURRENCY"
+)
+
+// MaxRetries limit upload and download retries during interaction with S3
+var MaxRetries = 15
```

Comment:
Keeping this magic const looks strange

Reply:
Okay

Reply:
Let's add this setting in a another PR? To keep scope of this clear.